### DISCUSSION TITLE
Create a new workflow for OnTenantCreated.

### DISCUSSION THREAD

**Discussion by JoshLefebvre** [OrchardCMS/OrchardCore#15999] [Q&A]
I have recently started working with workflows and I see a lot of potential for other areas that I can utilize them for in my current platform instead of coding custom logic. 

I wanted to create a new event for "TenantCreatedEvent" activity to help automate the tenant creation process. I realize there is already a  "CreateTenantTask"... but I am not quite sure that this is what I am looking for. 

Before I start developing I want to make sure there isn't anything that would match my current workflow:
1. Admin from the default tenant "Creates" a new tenant providing the DB connection string and recipe. I would need a new field for tenant admin (not super user)
2. OnTenantCreatedEvent Activity is fired
3. SendEmail Task activity sends the "Setup Url" (with the token) to the user's email from step 1.
4. The new user is responsible for filling out the setup page. I would like to extend this page with other site settings (such as site logo and site favicon using my custom settings library https://github.com/JoshLefebvre/OrchardCore.CustomStyleSettings). 

Please let me know if TenantCreatedEvent is something that the OC project would be interested in and I'd be happy to help develop it.

**Comment by agriffard** [comment]
See #5858 and #5878

**Comment by JoshLefebvre** [comment]
Nice! Yes I figure I will eventually need a tenant setup event in addition to the tenant created event. So it seems like this issue is put on hold temporarily? 

Looking at the tenant created logic, it seems much simpler. In addition to the validation (which perhaps should be centralized), it seems to only perform the following two operations:

```csharp
                    await _shellHost.UpdateShellSettingsAsync(shellSettings);
                    var token = CreateSetupToken(shellSettings);
                     return Ok(GetEncodedUrl(shellSettings, token));
```

I would only need the shellsettings  and the encoded url as output.
